### PR TITLE
Add DrivenMiles, Passiveness and total collision metrics

### PR DESCRIPTION
**Pull request by theDebugger811**

### CODE REVIEW

File: l5kit/l5kit/environment/callbacks.py [hunk 1]
```diff
@@ -83,6 +83,17 @@ def _on_step(self) -> bool:
                 self.logger.record(f'{self.prefix}/ade', ade_error)
                 self.logger.record(f'{self.prefix}/fde', fde_error)
 
+            # Calculate replayed/simulated driven miles
+            if 'replay_driven_miles' in self.metric_set.evaluation_plan.metrics_dict():
```
Comment (perone) [inline l5kit/l5kit/environment/callbacks.py:87]: I think we should check if we have these keys during the construction of this callback and trigger an error if the metric set doesn't have the required metrics/cms/etc.

File: l5kit/l5kit/environment/callbacks.py [hunk 2]
```diff
@@ -83,6 +83,17 @@ def _on_step(self) -> bool:
...
+                replay_driven_miles, simulated_driven_miles = self.compute_driven_miles()
+                self.logger.record(f'{self.prefix}/replay_driven_miles', replay_driven_miles)
+                self.logger.record(f'{self.prefix}/simulated_driven_miles', simulated_driven_miles)
+
+            # Calculate total collisions
+            if 'collision_front' in self.metric_set.evaluation_plan.metrics_dict():
```
Comment (perone) [inline l5kit/l5kit/environment/callbacks.py:93]: same here

File: l5kit/l5kit/environment/callbacks.py [hunk 3]
```diff
@@ -145,3 +156,34 @@ def compute_ade_fde(self) -> Tuple[float, float]:
         average_fde = sum(scene_fde_list) / len(scene_fde_list)
 
         return (average_ade, average_fde)
+
+    def compute_driven_miles(self) -> Tuple[float, float]:
```
Comment (perone) [inline l5kit/l5kit/environment/callbacks.py:160]: We do have a composite metric for computing driven miles, why not using that one ?

**Review by perone — COMMENTED** [review]
Thanks @theDebugger811, just a few comments.

File: l5kit/l5kit/environment/callbacks.py (hunk 3 above)
Comment (theDebugger811) [reply l5kit/l5kit/environment/callbacks.py]: Done!

**Review by perone — APPROVED** [review]

**Pull request merged**